src/services/resource.js

```diff
@@ -8,196 +8,580 @@
+
+		// get the unique organization ids from resource creator mapping table by the user
+		const OrganizationIds = utils.getUniqueElements(resourcesCreatedByMe.map((item) => item.organization_id))
+		const distinctInreviewResourceIds = await reviewsQueries.distinctResources(
+			{
+				organization_id: {
+					[Op.in]: OrganizationIds,
+				},
+				resource_id: {
+					[Op.in]: uniqueResourceIds,
+				},
+				status: common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+			},
+			['resource_id']
+		)
+
+		if (queryParams[common.STATUS] === common.REVIEW_STATUS_REQUESTED_FOR_CHANGES) {
+			filter = {
+				user_id: loggedInUserId,
+				organization_id: {
+					[Op.in]: OrganizationIds,
+				},
+				id: {
+					[Op.in]: distinctInreviewResourceIds.resource_ids,
+				},
 			}
-			let sort = {
-				sort_by: common.CREATED_AT,
-				order: common.SORT_DESC,
+		} else {
+			defaultFilter = {
+				user_id: loggedInUserId,
+				organization_id: {
+					[Op.in]: OrganizationIds,
+				},
+				id: {
+					[Op.in]: uniqueResourceIds,
+				},
+				status: {
+					[Op.in]: common.PAGE_STATUS_VALUES[common.PAGE_STATUS_SUBMITTED_FOR_REVIEW],
+				},
 			}
-			let filter = {
-				organization_id,
+
+			if (queryParams[common.STATUS]) {
+				defaultFilter.status = {
+					[Op.in]: queryParams[common.STATUS].split(','),
+				}
 			}
-			// fetch the details of resource and organization from resource creator mapping table by the user
-			const resource_creator_mapping_data = await resourceCreatorMappingQueries.findAll({ creator_id: user_id }, [
-				'resource_id',
-				'organization_id',
-			])
 
-			// get the unique resource ids from resource creator mapping table by the user
-			const uniqueResourceIds = [...new Set(resource_creator_mapping_data.map((item) => item.resource_id))]
+			filter = await this.constructCustomFilter(defaultFilter, queryParams, searchText)
+		}
 
-			// get the unique organization ids from resource creator mapping table by the user
-			const OrganizationIds = [...new Set(resource_creator_mapping_data.map((item) => item.organization_id))]
+		const sort = await this.constructSortOptions(queryParams)
+		const response = await resourceQueries.resourceList(
+			filter,
+			[
+				'id',
+				'title',
+				'organization_id',
+				'type',
+				'status',
+				'user_id',
+				'created_at',
+				'updated_at',
+				'submitted_on',
+				'published_on',
+				'last_reviewed_on',
+				'meta',
+			],
+			sort,
+			page,
+			limit
+		)
+		if (response.result.length <= 0) {
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'RESOURCE_LISTED_SUCCESSFULLY',
+				result,
+			})
+		}
 
-			if (queryParams[common.TYPE]) {
-				filter.type = {
-					[Op.in]: queryParams[common.TYPE].split(','),
+		const orgDetails = await this.fetchOrganizationDetails(
+			utils.getUniqueElements(response.result.map((item) => item.organization_id))
+		)
+		const commentMapping = await this.fetchOpenComments(
```

Comment:
@adithyadinesh0412 do groupBY by resource_id and use that instead of getting all data